arbnode/batch_poster.go

```diff
@@ -81,6 +82,8 @@ type BatchPosterConfig struct {
 	RedisUrl                           string                      `koanf:"redis-url"`
 	RedisLock                          SimpleRedisLockConfig       `koanf:"redis-lock" reload:"hot"`
 	ExtraBatchGas                      uint64                      `koanf:"extra-batch-gas" reload:"hot"`
+	UseSeparateL1Wallet                bool                        `koanf:"use-separate-l1-wallet"`
+	L1Wallet                           genericconf.WalletConfig    `koanf:"l1-wallet"`
```

Comment:
I'm wondering if we need `UseSeparateL1Wallet` or if we want to just use `L1Wallet` if it's non-null

Reply:
I dont think we can make L1Wallet a pointer since its inside koanf, though open to ideas on removing UseSeparateL1Wallet.

Reply:
Done.